### PR TITLE
add keystone authenticate

### PR DESCRIPTION
**Pull request by solofox**
Add authenticate support to java api. Keystone token is requested using openstack4j library.
usage:

``` java
_api = ApiConnectorFactory.build("192.168.1.150", 8082)
                .credentials("admin", "admin").tenantName("admin")
                .authServer("keystone", "http://192.168.1.150:5000/v2.0");
```

It would be better to register contrail api service to keystone, and Integrate ApiConnectorFactory with openstack4j's OSFactory:

```
OSClient os = OSFactory.builder()
                       .endpoint("http://127.0.0.1:5000/v2.0")
                       .credentials("admin","sample")
                       .tenantName("admin")
                       .authenticate();
_api = os.VncApi()
```

### CODE REVIEW

**Comment by vaidyasd** [comment]
This commit not config option to disable keystone authentication for java apis. e.g, in vmware vcenter only deployment, there is no keystone server to authenticate with. This code is broken.

**Comment by solofox** [comment]
@vaidyasd, I think nothing is broken.
1. In no keystone authentication environment, you can just create the api instance as before:
   _api = ApiConnectorFactory.build("192.168.1.150", 8082);
   **In other words, the authentication is disabled by default.**
2. The authentication only take effects when the api server give a 401 (no authenticate) response after a request, this behavior is same as python api.
   **In other words, the authentication is only enabled when api server require it.**

**Comment by solofox** [comment]
I have two servers: 208 has anonymous access, 200 need authentication, the test case gives following outputs. Both run without problems, nothing hangs.

```
Has  credential for 208(anonymous):
  Get 1 domains: default-domain
Miss credential for 208(anonymous):
  Get 1 domains: default-domain
Has  credential for 200(need-auth):
  Get 1 domains: default-domain
Miss credential for 200(need-auth):
  Server return null
```

test case source code:

```
    public static void main(String[] args) throws Exception {
        // TODO Auto-generated method stub
        ApiConnector api;
        List l;

        System.out.println("Has  credential for 208(anonymous): ");
        api = ApiConnectorFactory.build("192.168.1.208", 8082)
                .credentials("admin", "admin").tenantName("admin")
                .authServer("keystone", "http://192.168.1.208:5000/v2.0");
        l = api.list(Domain.class, null);
        if (l != null) {
            System.out.print("  Get " + l.size() + " domains:");
            for (int i = 0; i < l.size(); i++) {
                Domain d = (Domain)l.get(i);
                System.out.print(" " + d.getName());
            }
            System.out.println("");
        }
        else {
            System.out.println("  Server return null");
        }

        System.out.println("Miss credential for 208(anonymous): ");
        api = ApiConnectorFactory.build("192.168.1.208", 8082);
        l = api.list(Domain.class, null);
        if (l != null) {
            System.out.print("  Get " + l.size() + " domains:");
            for (int i = 0; i < l.size(); i++) {
                Domain d = (Domain)l.get(i);
                System.out.print(" " + d.getName());
            }
            System.out.println("");
        }
        else {
            System.out.println("  Server return null");
        }

        System.out.println("Has  credential for 200(need-auth): ");
        api = ApiConnectorFactory.build("192.168.1.200", 8082)
                .credentials("admin", "admin").tenantName("admin")
                .authServer("keystone", "http://192.168.1.200:5000/v2.0");
        l = api.list(Domain.class, null);
        if (l != null) {
            System.out.print("  Get " + l.size() + " domains:");
            for (int i = 0; i < l.size(); i++) {
                Domain d = (Domain)l.get(i);
                System.out.print(" " + d.getName());
            }
            System.out.println("");
        }
        else {
            System.out.println("  Server return null");
        }

        System.out.println("Miss credential for 200(need-auth): ");
        api = ApiConnectorFactory.build("192.168.1.200", 8082);
        l = api.list(Domain.class, null);
        if (l != null) {
            System.out.print("  Get " + l.size() + " domains:");
            for (int i = 0; i < l.size(); i++) {
                Domain d = (Domain)l.get(i);
                System.out.print(" " + d.getName());
            }
            System.out.println("");
        }
        else {
            System.out.println("  Server return null");
        }
    }
```
